### PR TITLE
Applied mutation testing to improve the quality of the tests and the code for Java implementation.

### PR DESCRIPTION
**Pull request by omgm**
Applied [mutation testing](http://confreaks.tv/videos/mwrc2014-re-thinking-regression-testing) with Ortask Mutator 1.9. Improved the quality of the tests (and the code) by 13% (from a score of 84.6% to 95.3%).
- Explicitly made public fields 'public'.
- Corrected the comparison of the error value by introducing bounds for it (added a new error precision field). 
  -- Best programming practice: floats/doubles should never be compared like ints.
- Made the new error precision part of the public API.
- Added needed tests to document behavior with bad and edge-case parameters.

### CODE REVIEW

File: java/src/main/java/me/everything/inbloom/BloomFilter.java [hunk 1]
```diff
@@ -24,11 +24,12 @@
     // These fields are part of the public interface of this structure.
     // Client code may read these values if desired. Client code MUST NOT
     // modify any of these.
-    int entries;
-    double error;
-    int bits;
-    int bytes;
-    int hashes;
+    public int entries;
```
Comment (bergundy) [inline java/src/main/java/me/everything/inbloom/BloomFilter.java]: If these are public, might as well make them final

File: java/src/main/java/me/everything/inbloom/BloomFilter.java [hunk 2]
```diff
@@ -24,11 +24,12 @@
...
-    double error;
-    int bits;
-    int bytes;
-    int hashes;
+    public int entries; 
+    public double error;
+    public int bits;
+    public int bytes;
+    public int hashes;
+    public static double errorPrecision = 0.000000001;
```
Comment (bergundy) [inline java/src/main/java/me/everything/inbloom/BloomFilter.java]: Makes sense, @dvirsky ?

File: java/src/main/java/me/everything/inbloom/BloomFilter.java (hunk 1 above)
Comment (dvirsky) [inline java/src/main/java/me/everything/inbloom/BloomFilter.java]: @bergundy they were package private before this PR. If the test is running in the same package, I suggest we keep them package-private, and final is just a nice addition.

File: java/src/main/java/me/everything/inbloom/BloomFilter.java (hunk 2 above)
Comment (dvirsky) [inline java/src/main/java/me/everything/inbloom/BloomFilter.java]: yes but it means changing this in all 3 implementations as well

File: java/src/test/java/me/everything/inbloom/BloomFilterTest.java [hunk 3]
```diff
@@ -9,6 +9,102 @@
  */
 public class BloomFilterTest extends TestCase {
 
+    public void testCreateFilterWithBadParameters()
+    {
+        try 
+        {
+            BloomFilter_mutated bf = new BloomFilter_mutated( -1, 0.01 );
```
Comment (bergundy) [inline java/src/test/java/me/everything/inbloom/BloomFilterTest.java]: Where does the BloomFilter_mutated class come from?
I don't understand

Comment (dvirsky) [inline java/src/test/java/me/everything/inbloom/BloomFilterTest.java]: I don't see `class BloomFilter_mutated` defined anywhere. Am I missing something?

File: java/src/test/java/me/everything/inbloom/BloomFilterTest.java [hunk 4]
```diff
@@ -9,6 +9,102 @@
  */
 public class BloomFilterTest extends TestCase {
 
+    public void testCreateFilterWithBadParameters()
+    {
```
Comment (bergundy) [inline java/src/test/java/me/everything/inbloom/BloomFilterTest.java]: I understand that there's no convention for braces in the project, I think we should go with omitting the new line before opening braces.
I'll fix in the rest of the project.

Comment (dvirsky) [inline java/src/test/java/me/everything/inbloom/BloomFilterTest.java]: @bergundy oh there is - no newline before opening brace

File: java/src/test/java/me/everything/inbloom/BloomFilterTest.java (hunk 3 above)
Comment (omgm) [inline java/src/test/java/me/everything/inbloom/BloomFilterTest.java]: Oops! An artifact of mutation analysis. I'll fix it shortly.

Comment (omgm) [inline java/src/test/java/me/everything/inbloom/BloomFilterTest.java]: Done. Let me know if you see the changes. 
Good catch btw! This was my own mistake by not running the tests on the final copy. Lesson learned.

File: java/src/main/java/me/everything/inbloom/BloomFilter.java (hunk 1 above)
Comment (bergundy) [inline java/src/main/java/me/everything/inbloom/BloomFilter.java]: let's go with protected

Comment (bergundy) [inline java/src/main/java/me/everything/inbloom/BloomFilter.java]: and final

Comment (bergundy) [inline java/src/main/java/me/everything/inbloom/BloomFilter.java]: @omgm please change the attributes to final protected

Comment (omgm) [inline java/src/main/java/me/everything/inbloom/BloomFilter.java]: Sorry guys, didn't know you were waiting on me. Changes have been done.

**Comment by bergundy** [comment]
@omgm merged, thanks!
